### PR TITLE
Add secure storage support

### PR DESCRIPTION
**Pull request by fahslaj**

### CODE REVIEW

File: src/main/java/cceclipseplugin/core/PluginManager.java [hunk 1]
```diff
@@ -179,19 +182,26 @@ public DataManager getDataManager() {
...
+				password[0] = secureStore.get(PreferenceConstants.PASSWORD, null);
+			} catch (StorageException e) {
+				e.printStackTrace();
+			}
 			IPreferenceStore prefStore = Activator.getDefault().getPreferenceStore();
-			String username = prefStore.getString(PreferenceConstants.USERNAME);
-			String password = prefStore.getString(PreferenceConstants.PASSWORD);
-			boolean showWelcomeDialog = (username == null || username.equals("") || password == null || password.equals(""));
+			boolean showWelcomeDialog = (username[0] == null || username[0].equals("") || password[0] == null || password[0].equals(""));
 			if (showWelcomeDialog) {
```
Comment (julie-is-late) [inline src/main/java/cceclipseplugin/core/PluginManager.java]: why not just do this in line?

**Review by julie-is-late — CHANGES_REQUESTED** [review]
we've talked this over on slack, and have a couple other changes that need to at least be looked into

**Comment by fahslaj** [comment]
I"ve fixed everything we talked about, including:

- preferences sticking above register dialog

- showing a dialog prompting the user for another attempt at logging in after a secure storage failure

- one line change mentioned above

**Review by julie-is-late — APPROVED** [review]

**Comment by julie-is-late** [comment]
this lgtm

**Pull request merged**